### PR TITLE
remove supertest

### PR DESCRIPTION
**Pull request by paed01**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,27 +12,68 @@ const saveToJar = require("./lib/saveToJar");
...
+  close() {
+    const server = this.server;
+    if (server) this.server = server.close();
+  }
+  _startHttpServer(requestListener) {
+    const server = http.createServer(requestListener);
+    return new Promise((resolve) => {
+      server.listen(0, () => {
+        return resolve(server);
+      });
```
Comment (jonaswalden) [inline index.js:68]:
```suggestion
      server.listen(0, () => resolve(server));
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 Changelog
 ---------
 
+# 11.1.0
+
+- Replace supertest with @bonniernews/httptest
```
Comment (jonaswalden) [inline CHANGELOG.md:6]: not true anymore

File: index.js [hunk 3]
```diff
@@ -2,37 +2,77 @@
...
+const kOrigin = Symbol.for("origin");
 
 class OriginResponse {
-  constructor(uri, response, originHost, protocol) {
-    this[responseSymbol] = response;
-    const status = this.status = response.statusCode;
-    this.ok = status >= 200 && status < 300;
-    this.headers = new Map(Object.entries(response.headers));
-    this.url = originHost ? `${protocol}//${originHost}${uri}` : response.request.url;
+  constructor(response, originHost, protocol) {
```
Comment (jonaswalden) [inline index.js:18]: all responses are now from node fetch. we're basically just overriding the url-property, might be a good use case for Proxy / Reflect.

File: index.js [hunk 4]
```diff
@@ -12,27 +12,68 @@ const saveToJar = require("./lib/saveToJar");
...
+    this[kResponse] = response;
+    const {path} = url.parse(response.url);
+    this.url = originHost ? `${protocol}//${originHost}${path}` : response.url;
+    this.status = response.status;
+    this.ok = response.ok;
+    response.headers.delete("_fl-origin");
```
[the remark this answers is not in the document]
Comment (paed01) [reply index.js:24]: don't matter really since it is removed before passing it to WebPage

[the remark this answers is not in the document]
Comment (paed01) [reply index.js:24]: Isn't the x-prefix is deprecated? - https://datatracker.ietf.org/doc/html/rfc6648

[the remark this answers is not in the document]
Comment (jonaswalden) [reply index.js]: looky there. good catch. 
just thinking that underscores in header names aren't common. 
maybe the `fl-` prefix is enough?

**Review by jonaswalden — APPROVED** [review]

**Pull request merged**